run-build-functions.sh

```diff
@@ -102,6 +105,34 @@ run_yarn() {
   fi
 
 
+  if [ "$NETLIFY_YARN_WORKSPACES" = "true" ]
+  then
+    echo "NETLIFY_YARN_WORKSPACES feature flag set"
+    # YARN_IGNORE_PATH will ignore the presence of a local yarn executable (i.e. yarn 2) and default
+    # to using the global one (which, for now, is alwasy yarn 1.x). See https://yarnpkg.com/configuration/yarnrc#ignorePath
+    if YARN_IGNORE_PATH=1 yarn workspaces info
+    then
+      local package_locations
+      # Extract all the packages and respective locations. .data will be a JSON object like
+      # {
+      #   "my-package-1": {
+      #     "location": "packages/blog-1",
+      #     "workspaceDependencies": [],
+      #     "mismatchedWorkspaceDependencies": []
+      #   },
+      #   (...)
+      # }
+      # We need to cache all the node_module dirs, or we'll always be installing them on each run
+      mapfile -t package_locations <<< "$(YARN_IGNORE_PATH=1 yarn --json workspaces info | jq -r '.data | fromjson | to_entries | .[].value.location')"
```

Comment:
Would it work to directly set `NETLIFY_JS_WORKSPACE_LOCATIONS` instead of `package_locations`? This would require declaring `NETLIFY_JS_WORKSPACE_LOCATIONS` as a file-level array variable. This would remove the need for the intermediary variable `package_locations`.

Reply:
Right now the cache logic and the respective usage of `NETLIFY_JS_WORKSPACE_LOCATIONS` is kind of isolated from the `run_yarn` function, I would say there's some use in keeping it that way? 🤔 That way we're keeping all the convoluted logic required to cache `workspaces` in the respective `restore_js_workspaces_cache`, `cache_js_workspaces` and `cache_node_modules`. The only thing the `run_yarn` is doing right is to pass the `locations` to the underlying functions ensuring it's decoupled from that. My concern is that if we end up spreading the `NETLIFY_JS_WORKSPACE_LOCATIONS` variable around we might end up with a tighter spaghetti bowl 😅

Reply:
Makes sense :+1: